hazelcast/src/test/java/com/hazelcast/test/annotation/TestProperties.java

```diff
@@ -0,0 +1,12 @@
+package com.hazelcast.test.annotation;
+
+import java.lang.annotation.Target;
+import java.lang.annotation.Retention;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.RetentionPolicy;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface TestProperties {
+    String maxThreadsCalculatorClass();
```

Comment:
No, maxThreads will be the same. What I suggest is to change signature of annotation to:

```java

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface TestProperties {
    Class<? extends MaxThreadsAware> maxThreadsCalculatorClass();
}
```
That way, you don't need to do `Class.forName()` from class-name and cast it etc..